### PR TITLE
94 bug menu generates too many random colors

### PR DESCRIPTION
**Pull request by rajtarn**
Fixed by generating colors when sites are added, not when they are selected.

### CODE REVIEW

**Review by AlgotE — APPROVED** [review]
Great fix, works as intended and faster for large number of files

**Review by NilsArnlund — APPROVED** [review]
Nice. Good work, it seems to be working well. I approve, however, we should probably wait for an answer form @ViktorAndersson99 about the changes in Charts.tsx before merge.

**Pull request merged**
